libs/core.lua

```diff
@@ -505,57 +307,76 @@ local function makeCore(config)
 
     local writer = miniz.new_writer()
 
-    log("importing", #path > 0 and path or fs.base, "highlight")
-    -- TODO: Find target relative to path and use that instead of just target
-    importPath(writer, fs, path, nil, { "!" .. target })
-
-    if meta.dependencies then
-      local deps = {}
-      local modulesDir = pathJoin(path, "deps")
-      writer:add("deps/", "")
-      core.processDeps(deps, fs, modulesDir, meta.dependencies)
-      for alias, dep in pairs(deps) do
-        local tag = dep.author .. '/' .. dep.name .. '@' .. dep.version
-        if dep.disk then
-          local name = "deps/" .. (dep.disk:match("([^/\\]+)$") or alias)
-          log("adding", tag .. ' (' .. dep.disk .. ')', "highlight")
-          importPath(writer, fs, path, name)
-        elseif dep.hash then
-          log("adding", tag, "highlight")
-          local tag = db.loadAs("tag", dep.hash)
-          if tag.type == "tree" then
-            importDb(writer, "deps/" .. alias, tag.object)
-          elseif tag.type == "blob" then
-            importBlob(writer, "deps/" .. alias .. ".lua", tag.object)
-          end
+    local function importEntry(hash, path, mode)
+      local kind, value = assert(db.loadAny(hash))
+      mode = mode or modes[kind]
+      if mode == modes.tree then
+        if path then
+          writer:add(path .. "/", "")
+        end
+        for i = 1, #value do
+          local entry = value[i]
+          local newPath = path and path .. '/' .. entry.name or entry.name
+          importEntry(entry.hash, newPath, entry.mode)
+        end
+      elseif modes.isFile(mode) then
+        local base = path:match("^(.*)%.lua$")
+        if base then
+          -- TODO: add an option to disable this in case you want uncompiled lua files.
+          log("compiling", path)
+          local fn = assert(loadstring(value, "bundle:" .. path))
+          value = string.dump(fn)
         end
+        writer:add(path, value, 9)
+      else
+        error("Can't handle entry for " .. path)
       end
     end
 
+    importEntry(rootHash)
+
     assert(uv.fs_write(fd, writer:finalize(), binSize))
     uv.fs_close(fd)
     assert(uv.fs_rename(tempFile, target))
     log("done building", target)
   end
 
-
-  function core.make(path, target)
-    local fs, newPath = vfs(path)
-    local meta = pkg.query(fs, newPath)
-    if not meta then
-      error("Not a package at: " .. path)
+  local function defaultTarget(meta)
+    local target = meta.target or meta.name:match("[^/]+$")
+    if jit.os == "Windows" then
```

Comment:
Actually, I'd rather not do this.  We already have a lot of local variables.  Too many cause performance issues in the jit vm.  The two checks for `jit.os` are not performance sensitive and it's pretty obvious what they do.